core/src/main/java/io/fabric8/maven/core/service/PortForwardService.java

```diff
@@ -0,0 +1,242 @@
+                        try {
+                            currentPortForward.close();
+                        } catch (Exception e) {}
+                    }
+                }
+            }
+        };
+
+        // Adding the current pod if present to the list
+        Pod newPod = getNewestPod(podSelector);
+        if (newPod != null) {
+            forwardedPods.add(newPod);
+        }
+
+
+        final Watch watch = kubernetes.pods().watch(new Watcher<Pod>() {
+
+            @Override
+            public void eventReceived(Action action, Pod pod) {
+                monitor.lock();
+                try {
+                    Pod newPod = getNewestPod(podSelector); // may be null
+                    forwardedPods.add(newPod);
+                    podChanged.signal();
+                } finally {
+                    monitor.unlock();
+                }
+            }
+
+            @Override
+            public void onClose(KubernetesClientException e) {
+                // don't care
+            }
+        });
+
+        forwarderThread.start();
+
+        final Closeable handle = new Closeable() {
+            @Override
+            public void close() throws IOException {
+                try {
+                    watch.close();
+                } catch (Exception e) {}
+                try {
+                    forwarderThread.interrupt();
+                    forwarderThread.join(15000);
+                } catch (Exception e) {}
+            }
+        };
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                try {
+                    handle.close();
+                } catch (Exception e) {
+                    // suppress
+                }
+            }
+        });
+
+        return handle;
+    }
+
+    private boolean podEquals(Pod pod1, Pod pod2) {
+        if (pod1 == pod2) {
+            return true;
+        }
+        if (pod1 == null || pod2 == null) {
+            return false;
+        }
+        return KubernetesHelper.getName(pod1).equals(KubernetesHelper.getName(pod2));
+    }
+
+    private Pod getNewestPod(LabelSelector selector) {
+        FilterWatchListDeletable<Pod, PodList, Boolean, Watch, Watcher<Pod>> pods =
+                KubernetesClientUtil.withSelector(kubernetes.pods(), selector, log);
+
+        Pod targetPod = null;
+        PodList list = pods.list();
+        if (list != null) {
+            List<Pod> items = list.getItems();
+            if (items != null) {
+                for (Pod pod : items) {
+                    PodStatusType status = KubernetesHelper.getPodStatus(pod);
+                    switch (status) {
+                    case WAIT:
+                    case OK:
+                        if (targetPod == null || (KubernetesHelper.isPodReady(pod) && KubernetesResourceUtil.isNewerResource(pod, targetPod))) {
+                            targetPod = pod;
```

Comment:
Ehehe, this time it's not a matter of preference, I've to scan all the pods to find the best one...
 ;)